src/envs/pybullet_blocks.py

```diff
@@ -0,0 +1,754 @@
+                                     params_space),
+                # Move up.
+                self._move_end_effector_relative_to_block(
+                    "MoveEndEffectorToAboveBlock", (0., 0., self.pick_z),
+                    ("rel", "rel", "abs"), types, params_space),
+            ])
+
+        types = self._Stack.types
+        params_space = self._Stack.params_space
+        self._Stack: ParameterizedOption = \
+            utils.LinearChainParameterizedOption("Stack",
+            [
+                # Move to above the block on which we will stack.
+                self._move_end_effector_relative_to_block(
+                    "MoveEndEffectorToAboveBlock", (0., 0., self.pick_z),
+                    ("rel", "rel", "abs"), types, params_space),
+                # Move down to place.
+                self._move_end_effector_relative_to_block(
+                    "MoveEndEffectorToPlace",
+                    (0., 0., self.block_size + self._place_offset_z),
+                    ("rel", "rel", "rel"), types, params_space),
+                # Open fingers.
+                self._change_fingers("OpenFingers", self.open_fingers, types,
+                                     params_space),
+                # Move up.
+                self._move_end_effector_relative_to_block(
+                    "MoveEndEffectorToAboveBlock", (0., 0., self.pick_z),
+                    ("rel", "rel", "abs"), types, params_space),
+            ])
+
+        types = self._PutOnTable.types
+        params_space = self._PutOnTable.params_space
+        self._PutOnTable: ParameterizedOption = \
+            utils.LinearChainParameterizedOption("PutOnTable",
+            [
+                # Move to above the block on which we will stack.
+                self._move_end_effector_relative_totable(
+                    "MoveEndEffectorToAboveTable", (0., 0., self.pick_z),
+                    ("rel", "rel", "abs"), types, params_space),
+                # Move down to place.
+                self._move_end_effector_relative_totable(
+                    "MoveEndEffectorToPlaceOnTable",
+                    (0., 0., self.block_size + self._place_offset_z),
+                    ("rel", "rel", "rel"), types, params_space),
+                # Open fingers.
+                self._change_fingers("OpenFingers", self.open_fingers, types,
+                                     params_space),
+                # Move up.
+                self._move_end_effector_relative_totable(
+                    "MoveEndEffectorToAboveTable", (0., 0., self.pick_z),
+                    ("rel", "rel", "abs"), types, params_space),
+            ])
+
+        # One-time initialization of pybullet assets. Note that this happens
+        # in __init__ because many instance attributes are created.
+        # Skip test coverage because GUI is too expensive to use in unit tests
+        # and cannot be used in headless mode.
```

Comment:
`__init__` is getting super long. I could imagine a lot of this pybullet stuff will be repeated for new environments as well, so might be worth breaking out at least to different instance methods for readability.

Reply:
Looking below maybe I can start moving things into a `predicators/src/pybullet_helpers` module or something when I start working on my pybullet stuff

Reply:
there's already a predicators/src/pybullet_utils.py where a lot of this stuff can be moved to. in the meantime though, why can't we just have the _initialize_pybullet() method like we usually do, called from __init__?

Reply:
@ronuchit how would we deal with all of the class instances?

Reply:
idgi, let's discuss on slack

Reply:
TIL that linting only complains with `attribute-defined-outside-init` when the attribute is modified elsewhere in the class